src/containers/community/CommunityPosts.js

```diff
@@ -58,42 +65,58 @@ class CommunityPosts extends React.Component {
+    let { location: { query }, dispatch, community, params: { id } } = this.props
     const { currentUser } = this.context
 
     return <div>
       {hasFeature(currentUser, COMMUNITY_SETUP_CHECKLIST) && canModerate(currentUser, community) &&
-        <CommunitySetup community={community}/>}
-      {isMember(currentUser, community) && <PostEditor community={community}/>}
-      {hasFeature(currentUser, REQUEST_TO_JOIN_COMMUNITY) && !isMember(currentUser, community) && <div className='request-to-join'>
-        You are not a member of this community. <a onClick={() => this.requestToJoin()}className='button'>Request to Join</a>
-      </div>}
-      <ConnectedPostList {...{subject, id, query}}/>
+        <CommunitySetup community={community} dispatch={dispatch} />}
+      {hasFeature(currentUser, IN_FEED_PROFILE_COMPLETION_MODULES) && isMember(currentUser, community) &&
+        <ProfileCompletionModules person={currentUser} />}
+      {isMember(currentUser, community) &&
+        <PostEditor community={community} />}
+      {hasFeature(currentUser, REQUEST_TO_JOIN_COMMUNITY) && !isMember(currentUser, community) &&
+        <div className='request-to-join'>
+          You are not a member of this community. <a onClick={() => this.requestToJoin()} className='button'>Request to Join</a>
+        </div>}
+      <ConnectedPostList {...{subject, id, query}} />
       {!isMember(currentUser, community) && <div className='post-list-footer'>
         You are not a member of this community, so you are shown only posts that are marked as public.
       </div>}
     </div>
   }
 }
 
-export default compose(
-  prefetch(({ dispatch, params: { id }, query, currentUser, store }) =>
-    dispatch(fetch(subject, id, query))),
-  connect((state, { params }) => ({
-    community: state.communities[params.id],
-    currentUser: state.people.current
-  }))
-)(CommunityPosts)
-
-const CommunitySetup = connect()(({ community, dispatch }) => {
+const CommunitySetup = ({ community, dispatch }) => {
   const checklist = getChecklist(community)
   const percent = checklistPercentage(checklist)
-
   if (percent === 100) return null
-
   return <div className='community-setup'
     onClick={() => dispatch(showModal('checklist'))}>
-    <PercentBar percent={percent}/>
+    <PercentBar percent={percent} />
     Your community is {percent}% set up. <a>Click here</a> to continue setting it up.
   </div>
+}
+
+const ProfileCompletionModules = ({ person }) => {
+  if (!hasBio(person)) {
+    return <ProfileBioModule person={person} />
+  } else if (!hasSkills(person)) {
+    return <ProfileSkillsModule person={person} />
+  } else {
+    return null
+  }
+}
+
+// Redux connection
+
+const mapStateToProps = (state, { params }) => ({
+  community: state.communities[params.id],
+  currentUser: state.people.current
 })
+
+export default compose(
+  prefetch(({ dispatch, params: { id }, query, currentUser, store }) =>
```

Comment:
i know you just moved this from above, didn't write it yourself, but please remove the unused arguments